Mathlib/Algebra/Lie/Semisimple.lean

```diff
@@ -120,31 +111,470 @@ theorem hasTrivialRadical_iff_no_abelian_ideals :
+  have hs' : s' ⊂ s := Finset.erase_ssubset hJs
+  have hs'S : ↑s' ⊆ S := Set.Subset.trans (Finset.coe_subset.mpr hs'.subset) hs
+  -- If we show that `I` is contained in the supremum `K` of the complement of `J` in `s`,
+  -- then we are done by recursion.
+  set K := s'.sup id
+  suffices I ≤ K by
+    obtain ⟨t, hts', htI⟩ := atomistic_of_finset s' hs'S I this
+    exact ⟨t, le_trans hts' hs'.subset, htI⟩
+  -- Since `I` is contained in the supremum of `J` with the supremum of `s'`,
+  -- any element `x` of `I` can be written as `y + z` for some `y ∈ J` and `z ∈ K`.
+  intro x hx
+  obtain ⟨y, hy, z, hz, rfl⟩ : ∃ y ∈ id J, ∃ z ∈ K, y + z = x := by
+    rw [← LieSubmodule.mem_sup, ← Finset.sup_insert, Finset.insert_erase hJs]
+    exact hI.le hx
+  -- If we show that `y` is contained in the center of `J`,
+  -- then we find `x = z`, and hence `x` is contained in the supremum of `s'`.
+  -- Since `x` was arbitrary, we have shown that `I` is contained in the supremum of `s'`.
+  suffices ⟨y, hy⟩ ∈ LieAlgebra.center R J by
+    have _inst := hS.isSimple J (hs hJs)
+    rw [HasTrivialRadical.center_eq_bot R J, LieSubmodule.mem_bot] at this
+    apply_fun Subtype.val at this
+    dsimp at this
+    rwa [this, zero_add]
+  -- To show that `y` is in the center of `J`,
+  -- we show that any `j ∈ J` brackets to `0` with `z` and with `x = y + z`.
+  -- By a simple computation, that implies `⁅j, y⁆ = 0`, for all `j`, as desired.
+  intro j
+  suffices ⁅(j : L), z⁆ = 0 ∧ ⁅(j : L), y + z⁆ = 0 by
+    rw [lie_add, this.1, add_zero] at this
+    ext
+    exact this.2
+  rw [← LieSubmodule.mem_bot (R := R) (L := L), ← LieSubmodule.mem_bot (R := R) (L := L)]
+  constructor
+  -- `j` brackets to `0` with `z`, since `⁅j, z⁆` is contained in `⁅J, K⁆ ≤ J ⊓ K`,
+  -- and `J ⊓ K = ⊥` by the independence of the atoms.
+  · apply (hS.setIndependent.disjoint_sSup (hs hJs) hs'S (Finset.not_mem_erase _ _)).le_bot
+    apply LieSubmodule.lie_le_inf
+    apply LieSubmodule.lie_mem_lie _ _ j.2
+    simpa only [K, Finset.sup_id_eq_sSup] using hz
+  -- By similar reasoning, `j` brackets to `0` with `x = y + z ∈ I`, if we show `J ⊓ I = ⊥`.
+  suffices J ⊓ I = ⊥ by
+    apply this.le
+    apply LieSubmodule.lie_le_inf
+    exact LieSubmodule.lie_mem_lie _ _ j.2 hx
+  -- Indeed `J ⊓ I = ⊥`, since `J` is an atom that is not contained in `I`.
+  apply ((hS.isAtom J (hs hJs)).le_iff.mp _).resolve_right
+  · contrapose! hJI
+    rw [← hJI]
+    exact inf_le_right
+  exact inf_le_left
+termination_by s => s.card
+decreasing_by
+  simp_wf
+  exact Finset.card_lt_card hs'
+
+/--
+In a semisimple Lie algebra,
+Lie ideals that are contained in the supremum of a collection of atoms
+are themselves the supremum of a subcollection of those atoms.
+
+The proof is by a compactness argument, reducing to finite collections of atoms.
+In the finite case, the proof is given in `atomistic_of_finset`.
+-/
+lemma atomistic (J : LieIdeal R L) (hJ : J ≤ sSup S) : ∃ T ⊆ S, J = sSup T := by
```

Comment:
Is this possibly a result about compactly-generated modular lattices, a rather than about the lattice `LieIdeal R L`?

For example I noticed a while ago that independence in such lattices can be characterised just by working with finite sets so I added `CompleteLattice.independent_iff_supIndep_of_injOn` (in #8154)

I admit I haven't read this proof closely so I may be wrong but superficially this looks like a similar result except for being atomistic rather than independent.

Reply:
I tried my best at abstracting this.